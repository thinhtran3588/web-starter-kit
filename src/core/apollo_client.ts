import ApolloClient, { gql } from 'apollo-boost';
import fetch from 'isomorphic-unfetch';
import { config } from '@app/config';

export const apolloClient = new ApolloClient({
  uri: config.apiEndpoint,
  fetch,
});
export { gql };
