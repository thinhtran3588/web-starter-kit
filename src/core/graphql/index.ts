export * from './queries/getCurrentUserQuery';
export * from './queries/getCurrentNotificationQuery';
export * from './apolloClient';
