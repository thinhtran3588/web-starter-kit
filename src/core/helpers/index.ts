export * from './format_with_lookup';
export * from './format_date';
export * from './format_date_time';
export * from './show_notification';
export * from './catch_error';
export * from './handle_error';
export * from './app_error';
export * from './sanitize_form_data';
export * from './get_error_message';
export * from './sleep';
