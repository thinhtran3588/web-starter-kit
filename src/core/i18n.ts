import NextI18Next, {
  WithTranslation as NextI18NextWithTranslation,
  TFunction as NextI18NexTFunction,
} from 'next-i18next';
import { config } from '@app/config';

const otherLanguages = config.i18n.languages
  .filter((language) => language.code !== config.i18n.defaultLang)
  .map((language) => language.code);
const localeSubpaths: { [code: string]: string } = {};
otherLanguages.forEach((language) => {
  localeSubpaths[language] = language;
});
export const nextI18next = new NextI18Next({
  defaultLanguage: config.i18n.defaultLang,
  otherLanguages,
  localeSubpaths,
});

/* Optionally, export class methods as named exports */
export const { appWithTranslation, withTranslation, i18n } = nextI18next;
export type WithTranslation = NextI18NextWithTranslation;
export type TFunction = NextI18NexTFunction;
