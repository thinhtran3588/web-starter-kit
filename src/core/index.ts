export * from './theme';
export * from './interfaces/nav_item';
export * from './interfaces/offset_pagination';
export * from './interfaces/offset_pagination_result';
export * from './interfaces/field_info';
export * from './interfaces/filter';
export * from './interfaces/filte_with_offset_pagination';
export * from './interfaces/picker_data_item';
export * from './interfaces/field_type';
export * from './interfaces/field_value_type';
export * from './interfaces/table_column';
export * from './interfaces/next_request';
export * from './interfaces/search_record';
export * from './i18n';
export * from './apollo_client';
