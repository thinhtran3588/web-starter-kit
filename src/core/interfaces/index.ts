export * from './nav_item';
export * from './offset_pagination';
export * from './offset_pagination_result';
export * from './field_info';
export * from './filter';
export * from './filte_with_offset_pagination';
export * from './picker_data_item';
export * from './field_type';
export * from './field_value_type';
export * from './table_column';
export * from './next_request';
export * from './search_record';
export * from './user';
export * from './login_type';
export * from './form_data';
export * from './notification_content';
export * from './country';
export * from './error_with_code';
