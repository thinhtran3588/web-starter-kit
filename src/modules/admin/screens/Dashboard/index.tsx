import React from 'react';
import { AdminLayout } from '@app/components';
import { useStyles } from './styles';

export const Dashboard = (): JSX.Element => {
  const classes = useStyles();
  return (
    <AdminLayout title='Dashboard' description='Dashboard'>
      <h1 className={classes.header}>Dashboard</h1>
    </AdminLayout>
  );
};
