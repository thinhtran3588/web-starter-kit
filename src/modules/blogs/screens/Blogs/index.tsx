import React from 'react';
import { Grid } from '@material-ui/core';
import { Layout } from '@app/components';
import { Categories, BlogsList } from './components';

export const Blogs = (): JSX.Element => {
  return (
    <Layout title='Blogs' description='Blogs'>
      <Grid container>
        <Grid item xs={1} lg={2}></Grid>
        <Grid item xs={10} lg={8}>
          <Categories />
          <BlogsList />
        </Grid>
        <Grid item xs={1} lg={2}></Grid>
      </Grid>
    </Layout>
  );
};
