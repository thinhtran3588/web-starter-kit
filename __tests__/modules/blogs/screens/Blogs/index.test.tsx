import React from 'react';
import { render } from '@testing-library/react';
import { Blogs } from '@app/modules/blogs/screens/Blogs';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core';

describe('@app/modules/blogs/screens/Blogs', () => {
  const theme = createMuiTheme({
    props: {
      MuiWithWidth: {
        initialWidth: 'xs',
      },
    },
  });
  beforeEach(() => {});

  it('renders successfully', async () => {
    const { baseElement } = render(
      <MuiThemeProvider theme={theme}>
        <Blogs />
      </MuiThemeProvider>,
    );
    expect(baseElement).toMatchSnapshot();
  });
});
