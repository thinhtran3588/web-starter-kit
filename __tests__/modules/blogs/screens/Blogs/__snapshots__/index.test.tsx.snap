// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@app/modules/blogs/screens/Blogs renders successfully 1`] = `
<body>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-root-3 MuiAppBar-colorPrimary mui-fixed"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <next-link
          href="/"
        >
          <a
            class=""
            color="inherit"
            href="/"
            title="Home"
          >
            <img
              alt="Logo"
              class="makeStyles-logo-5"
              src="/static/images/logo.svg"
            />
          </a>
        </next-link>
        <div
          class="makeStyles-flexGrow-4"
        />
        <div>
          <button
            aria-label="common.changeLanguage"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <div
                class="MuiAvatar-root makeStyles-icon-52 MuiAvatar-colorDefault"
              >
                EN
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <button
          aria-label="notification"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <span
              class="MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                />
              </svg>
              <span
                class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary MuiBadge-dot"
              />
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          data-testid="menu-icon"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <div
      class="makeStyles-container-1"
    >
      <main
        class="makeStyles-main-2"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-lg-2"
          />
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-lg-8"
          >
            <div
              class="makeStyles-categoriesContainer-212"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-6"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <div
                        class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-213 MuiPaper-rounded"
                      >
                        <div>
                          <h2
                            class="makeStyles-categoriesTitle-214"
                          >
                            // This impressive paella
                          </h2>
                          <p
                            class="makeStyles-categoriesDescription-215"
                          >
                            Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Vestibulum ac diam sit amet quam vehicula elementum sed sit amet dui
                          </p>
                        </div>
                        <div
                          class="makeStyles-categoriesButtonContainer-216"
                        >
                          <button
                            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="MuiButton-label"
                            >
                              Explore
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </button>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <div
                        class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card2-217 MuiPaper-rounded"
                      >
                        <div>
                          <h2
                            class="makeStyles-categoriesTitle-214"
                          >
                            // This impressive paella
                          </h2>
                          <p
                            class="makeStyles-categoriesDescription-215"
                          >
                            Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Vestibulum ac diam sit amet quam vehicula elementum sed sit amet dui
                          </p>
                        </div>
                        <div
                          class="makeStyles-categoriesButtonContainer-216"
                        >
                          <button
                            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="MuiButton-label"
                            >
                              Explore
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-3"
                >
                  <div
                    class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card3-218 MuiPaper-rounded"
                  >
                    <div>
                      <h2
                        class="makeStyles-categoriesTitle-214"
                      >
                        // This impressive paella
                      </h2>
                      <p
                        class="makeStyles-categoriesDescription-215"
                      >
                        Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Vestibulum ac diam sit amet quam vehicula elementum sed sit amet dui
                      </p>
                    </div>
                    <div
                      class="makeStyles-categoriesButtonContainer-216"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          Explore
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-3"
                >
                  <div
                    class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card4-219 MuiPaper-rounded"
                  >
                    <div>
                      <h2
                        class="makeStyles-categoriesTitle-214"
                      >
                        // This impressive paella
                      </h2>
                      <p
                        class="makeStyles-categoriesDescription-215"
                      >
                        Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Vestibulum ac diam sit amet quam vehicula elementum sed sit amet dui
                      </p>
                    </div>
                    <div
                      class="makeStyles-categoriesButtonContainer-216"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          Explore
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="makeStyles-blogsListContainer-250"
            >
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-249 MuiTypography-colorPrimary"
                href="/blogs/nulla-porttitor-accumsan-tincidunt"
                target="_blank"
              >
                <div
                  class="MuiGrid-root makeStyles-grid-252 MuiGrid-container"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
                  >
                    <div
                      class="makeStyles-blogImage-253"
                    />
                  </div>
                  <div
                    class="MuiGrid-root makeStyles-bloginContentContainer-254 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
                  >
                    <div>
                      <h2
                        class="makeStyles-blogTitle-255"
                      >
                        Nulla porttitor accumsan tincidunt. Quisque velit nisi, pretium ut lacinia in, elementum id enim.
                      </h2>
                      <p
                        class="makeStyles-blogDescription-256"
                      >
                        Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Donec velit neque, auctor sit amet aliquam vel, ullamcorper sit amet ligula. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Donec velit neque, auctor sit amet aliquam vel, ullamcorper sit amet ligula. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem.
                      </p>
                    </div>
                    <div
                      class="makeStyles-blogItemFooter-258"
                    >
                      <div>
                        Author: 
                        <span
                          class="makeStyles-boldText-257"
                        >
                          Lorem ipsum
                        </span>
                         | Category:
                         
                        <span
                          class="makeStyles-boldText-257"
                        >
                          Vivamus suscipit
                        </span>
                      </div>
                      <div>
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiButton-label"
                          >
                            See more
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </a>
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-249 MuiTypography-colorPrimary"
                href="/blogs/nulla-porttitor-accumsan-tincidunt"
                target="_blank"
              >
                <div
                  class="MuiGrid-root makeStyles-grid-252 MuiGrid-container"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
                  >
                    <div
                      class="makeStyles-blogImage-253"
                    />
                  </div>
                  <div
                    class="MuiGrid-root makeStyles-bloginContentContainer-254 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
                  >
                    <div>
                      <h2
                        class="makeStyles-blogTitle-255"
                      >
                        Nulla porttitor accumsan tincidunt. Quisque velit nisi, pretium ut lacinia in, elementum id enim.
                      </h2>
                      <p
                        class="makeStyles-blogDescription-256"
                      >
                        Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Donec velit neque, auctor sit amet aliquam vel, ullamcorper sit amet ligula. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Donec velit neque, auctor sit amet aliquam vel, ullamcorper sit amet ligula. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem.
                      </p>
                    </div>
                    <div
                      class="makeStyles-blogItemFooter-258"
                    >
                      <div>
                        Author: 
                        <span
                          class="makeStyles-boldText-257"
                        >
                          Lorem ipsum
                        </span>
                         | Category:
                         
                        <span
                          class="makeStyles-boldText-257"
                        >
                          Vivamus suscipit
                        </span>
                      </div>
                      <div>
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiButton-label"
                          >
                            See more
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </a>
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-249 MuiTypography-colorPrimary"
                href="/blogs/nulla-porttitor-accumsan-tincidunt"
                target="_blank"
              >
                <div
                  class="MuiGrid-root makeStyles-grid-252 MuiGrid-container"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
                  >
                    <div
                      class="makeStyles-blogImage-253"
                    />
                  </div>
                  <div
                    class="MuiGrid-root makeStyles-bloginContentContainer-254 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
                  >
                    <div>
                      <h2
                        class="makeStyles-blogTitle-255"
                      >
                        Nulla porttitor accumsan tincidunt. Quisque velit nisi, pretium ut lacinia in, elementum id enim.
                      </h2>
                      <p
                        class="makeStyles-blogDescription-256"
                      >
                        Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Donec velit neque, auctor sit amet aliquam vel, ullamcorper sit amet ligula. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Donec velit neque, auctor sit amet aliquam vel, ullamcorper sit amet ligula. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem.
                      </p>
                    </div>
                    <div
                      class="makeStyles-blogItemFooter-258"
                    >
                      <div>
                        Author: 
                        <span
                          class="makeStyles-boldText-257"
                        >
                          Lorem ipsum
                        </span>
                         | Category:
                         
                        <span
                          class="makeStyles-boldText-257"
                        >
                          Vivamus suscipit
                        </span>
                      </div>
                      <div>
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiButton-label"
                          >
                            See more
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </a>
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-249 MuiTypography-colorPrimary"
                href="/blogs/nulla-porttitor-accumsan-tincidunt"
                target="_blank"
              >
                <div
                  class="MuiGrid-root makeStyles-grid-252 MuiGrid-container"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
                  >
                    <div
                      class="makeStyles-blogImage-253"
                    />
                  </div>
                  <div
                    class="MuiGrid-root makeStyles-bloginContentContainer-254 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
                  >
                    <div>
                      <h2
                        class="makeStyles-blogTitle-255"
                      >
                        Nulla porttitor accumsan tincidunt. Quisque velit nisi, pretium ut lacinia in, elementum id enim.
                      </h2>
                      <p
                        class="makeStyles-blogDescription-256"
                      >
                        Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Donec velit neque, auctor sit amet aliquam vel, ullamcorper sit amet ligula. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Donec velit neque, auctor sit amet aliquam vel, ullamcorper sit amet ligula. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem.
                      </p>
                    </div>
                    <div
                      class="makeStyles-blogItemFooter-258"
                    >
                      <div>
                        Author: 
                        <span
                          class="makeStyles-boldText-257"
                        >
                          Lorem ipsum
                        </span>
                         | Category:
                         
                        <span
                          class="makeStyles-boldText-257"
                        >
                          Vivamus suscipit
                        </span>
                      </div>
                      <div>
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiButton-label"
                          >
                            See more
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </a>
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-249 MuiTypography-colorPrimary"
                href="/blogs/nulla-porttitor-accumsan-tincidunt"
                target="_blank"
              >
                <div
                  class="MuiGrid-root makeStyles-grid-252 MuiGrid-container"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
                  >
                    <div
                      class="makeStyles-blogImage-253"
                    />
                  </div>
                  <div
                    class="MuiGrid-root makeStyles-bloginContentContainer-254 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
                  >
                    <div>
                      <h2
                        class="makeStyles-blogTitle-255"
                      >
                        Nulla porttitor accumsan tincidunt. Quisque velit nisi, pretium ut lacinia in, elementum id enim.
                      </h2>
                      <p
                        class="makeStyles-blogDescription-256"
                      >
                        Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Donec velit neque, auctor sit amet aliquam vel, ullamcorper sit amet ligula. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Donec velit neque, auctor sit amet aliquam vel, ullamcorper sit amet ligula. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem.
                      </p>
                    </div>
                    <div
                      class="makeStyles-blogItemFooter-258"
                    >
                      <div>
                        Author: 
                        <span
                          class="makeStyles-boldText-257"
                        >
                          Lorem ipsum
                        </span>
                         | Category:
                         
                        <span
                          class="makeStyles-boldText-257"
                        >
                          Vivamus suscipit
                        </span>
                      </div>
                      <div>
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiButton-label"
                          >
                            See more
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </a>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-lg-2"
          />
        </div>
      </main>
      <footer
        class="makeStyles-root-295"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-lg-2"
          />
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-lg-8"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-3"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6"
                >
                  common.company
                </h6>
                <div
                  class="makeStyles-linkContainer-297"
                >
                  <next-link
                    href="/"
                  >
                    <a
                      class="makeStyles-link-296"
                      color="inherit"
                      href="/"
                    >
                      common.home
                    </a>
                  </next-link>
                </div>
                <div
                  class="makeStyles-linkContainer-297"
                >
                  <next-link
                    href="/aboutUs"
                  >
                    <a
                      class="makeStyles-link-296"
                      color="inherit"
                      href="/aboutUs"
                    >
                      common.aboutUs
                    </a>
                  </next-link>
                </div>
                <div
                  class="makeStyles-linkContainer-297"
                >
                  <next-link
                    href="/blogs"
                  >
                    <a
                      class="makeStyles-link-296"
                      color="inherit"
                      href="/blogs"
                    >
                      common.blogs
                    </a>
                  </next-link>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6"
                >
                  common.socialMedias
                </h6>
                <div>
                  <next-link
                    href="/"
                  >
                    <a
                      class="makeStyles-socialLink-298"
                      color="inherit"
                      href="/"
                    >
                      <img
                        class="makeStyles-image-299"
                        src="/static/images/facebook.png"
                      />
                    </a>
                  </next-link>
                  <next-link
                    href="/"
                  >
                    <a
                      class="makeStyles-socialLink-298"
                      color="inherit"
                      href="/"
                    >
                      <img
                        class="makeStyles-image-299"
                        src="/static/images/linkedin.png"
                      />
                    </a>
                  </next-link>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6"
                >
                  ©
                   
                  <next-link
                    href="/"
                  >
                    <a
                      class="makeStyles-link-296"
                      color="inherit"
                      href="/"
                    >
                      Thinh Tran
                    </a>
                  </next-link>
                   
                  2019
                </h6>
                <span
                  class="MuiTypography-root MuiTypography-caption"
                >
                  common.footer.createdWithLove
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-lg-2"
          />
        </div>
      </footer>
    </div>
  </div>
  <div
    aria-hidden="true"
    class="MuiPopover-root"
    id="simple-menu"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px; visibility: hidden;"
  >
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiMenu-paper MuiPaper-elevation8 MuiPopover-paper MuiPaper-rounded"
      style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
      tabindex="-1"
    >
      <ul
        class="MuiList-root MuiMenu-list MuiList-padding"
        role="menu"
        tabindex="-1"
      >
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="0"
        >
          English
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="-1"
        >
          Tiếng Việt
          <span
            class="MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
